### PR TITLE
Extract types

### LINKED ISSUE #11: Extracting reusable types to their own interfaces
I suggest extracting reusable types to their own interfaces to improve the reusability and maintainability of our codebase.

### Proposed Changes
- Identify common types that are used across multiple components or modules
- Extract these types to their own interfaces
- Update existing code to use the new interfaces

### Additional Information
- This change will improve code reusability and maintainability
- We can avoid duplicating code by sharing interfaces across files

### Tasks
- [x] Identify common types to extract
- [x] Create new interfaces for these types
- [x] Update existing components and modules to use the new interfaces

### PR DESCRIPTION
**Pull request by BilligsterUser**
closes #11

### CODE REVIEW

**Comment by BilligsterUser** [comment]
Naming things is hard 😅

File: src/CashuMint.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import axios from "axios";
-import { Proof } from "./model/Proof.js";
-import { SerealizedBlindedSignature } from "./model/types/SerealizedBlinedSignature.js";
+import {
```
Comment (gandlafbtc) [inline src/CashuMint.ts:2]: Man, sorry it's my fault, but it should be serialized, with i instead of e

Comment (gandlafbtc) [reply src/CashuMint.ts:2]: https://en.m.wikipedia.org/wiki/Serialization

can also use a different prefix altogether, don't know  how fitting it is.

File: src/model/types/index.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { Proof } from '../Proof.js'
```
Comment (gandlafbtc) [inline src/model/types/index.ts:1]: nice! that's better here.

Before we merge, we could add types for the payloads in here

**Review by gandlafbtc — COMMENTED** [review]
🔥

File: src/model/types/index.ts (hunk 2 above)
Comment (BilligsterUser) [reply src/model/types/index.ts:1]: I added a CheckSpendablePayload type I don't think there is much sense to add types for the other payloads.

**Pull request merged**